pkg/config_file.go

```diff
@@ -13,66 +20,119 @@ type FileConfig struct {
-					Timeout:               timeout,
-					RequestRepeats:        requestRepeats,
-					AllowRedirect:         allowRedirect,
-					AllowCodeBlocks:       allowCodeBlocks,
-					IgnoreExternal:        ignoreExternal,
-					IgnoreInternal:        ignoreInternal,
-				}
-			}
+		cfg := *config
+		file := File{}
+
+		if found, foundFile := findFile(filePath, cfg.Files); found {
+			file = foundFile
+		}
+
+		fileCfg := FileConfig{}
+		if file.Config != nil {
+			fileCfg = *file.Config
 		}
+
+		timeout := getDefaultIntIfNil(cfg.Timeout, fileCfg.Timeout)
+		requestRepeats := getDefaultIntIfNil(cfg.RequestRepeats, fileCfg.RequestRepeats)
+		allowRedirect := getDefaultBoolIfNil(cfg.AllowRedirect, fileCfg.AllowRedirect)
+		allowCodeBlocks := getDefaultBoolIfNil(cfg.AllowCodeBlocks, fileCfg.AllowCodeBlocks)
+
+		ignoreInternal := getInternalIgnorePolicy(filePath, cfg, file.Config)
+		ignoreExternal := getDefaultBoolIfNil(cfg.IgnoreExternal, fileCfg.IgnoreExternal)
+
+		externalLinksToIgnore := getExternalLinksToIgnore(cfg, file.Config)
+		internalLinksToIgnore := getInternalLinksToIgnore(cfg, file.Config)
+
 		return &FileConfig{
-			ExternalLinksToIgnore: config.ExternalLinksToIgnore,
-			InternalLinksToIgnore: config.InternalLinksToIgnore,
-			Timeout:               &config.Timeout,
-			RequestRepeats:        &config.RequestRepeats,
-			AllowRedirect:         &config.AllowRedirect,
-			AllowCodeBlocks:       &config.AllowCodeBlocks,
-			IgnoreExternal:        &config.IgnoreExternal,
-			IgnoreInternal:        &config.IgnoreInternal,
+			BasePath:              config.BasePath,
+			ExternalLinksToIgnore: externalLinksToIgnore,
+			InternalLinksToIgnore: internalLinksToIgnore,
+			Timeout:               &timeout,
+			RequestRepeats:        &requestRepeats,
+			AllowRedirect:         &allowRedirect,
+			AllowCodeBlocks:       &allowCodeBlocks,
+			IgnoreExternal:        &ignoreExternal,
+			IgnoreInternal:        &ignoreInternal,
 		}
 	}
 	return nil
 }
+
+func findFile(filePath string, files []File) (bool, File) {
+	for _, file := range files {
+		if filePath == file.RelPath && file.Config != nil {
+			return true, file
+		}
+	}
+	return false, File{}
+}
+
+func getExternalLinksToIgnore(config Config, fileConfig *FileConfig) []string {
+	externalLinksToIgnore := config.ExternalLinksToIgnore
+	if fileConfig != nil {
+		externalLinksToIgnore = unique(append(config.ExternalLinksToIgnore, fileConfig.ExternalLinksToIgnore...))
+	}
+
+	return externalLinksToIgnore
+}
+
+func getInternalLinksToIgnore(config Config, fileConfig *FileConfig) []string {
+	internalLinksToIgnore := config.InternalLinksToIgnore
+	if fileConfig != nil {
+		internalLinksToIgnore = unique(append(config.InternalLinksToIgnore, fileConfig.InternalLinksToIgnore...))
+	}
+
+	return internalLinksToIgnore
+}
+
+func getInternalIgnorePolicy(filepath string, config Config, fileConfig *FileConfig) bool {
+	internalIgnore := config.IgnoreExternal
```

Comment:
```suggestion
	internalIgnore := config.IgnoreInternal
```

Please cover this case in unit tests